### PR TITLE
MULE-8384: Removing new properties map creation and serializable propert...

### PR DESCRIPTION
**Pull request by afelisatti**
...ies removal in session merge, updating existing map with new properties instead.

### CODE REVIEW

File: core/src/main/java/org/mule/session/DefaultMuleSession.java [hunk 1]
```diff
@@ -217,19 +217,9 @@ public void merge(MuleSession updatedSession)
...
             return;
         }
-        Map<String, Object> oldProperties = this.properties;
-        this.properties = Collections.synchronizedMap(new CaseInsensitiveHashMap/* <String, Object> */());
-        for (String propertyKey : updatedSession.getPropertyNamesAsSet())
-        {
-            this.properties.put(propertyKey, updatedSession.<Object> getProperty(propertyKey));
-        }
-        for (Map.Entry<String, Object> property : oldProperties.entrySet())
+        for(String updatedPropertyKey : updatedSession.getPropertyNamesAsSet())
```
Comment (marcosnc) [inline core/src/main/java/org/mule/session/DefaultMuleSession.java]: Apply code style.

File: tests/integration/src/test/resources/foreach-session-var-config.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        <set-payload value="#[ [ 'A', 'B' ]  ]"/>
+        <set-session-variable variableName="mySessionList" value="#[groovy: new ArrayList()]"/>
+
+        <foreach>
+            <vm:outbound-endpoint exchange-pattern="request-response" path="out" />
+            <object-to-string-transformer />
+            <expression-transformer expression="#[sessionVars['mySessionList'].add(payload)]"/>
+        </foreach>
+
+        <logger level="ERROR" message="After second foreach: #[sessionVars['mySessionList']]" />
```
Comment (marcosnc) [inline tests/integration/src/test/resources/foreach-session-var-config.xml]: Remove logger from test.

File: tests/integration/src/test/java/org/mule/test/routing/ForEachSessionVarTestCase.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    public void testSessionVars() throws Exception
+    {
+        Collection<String> expectedArray = new ArrayList<>();
+        expectedArray.add("Hello World A");
+        expectedArray.add("Hello World B");
+        muleContext.getClient().send("vm://in", getTestMuleMessage());
+        assertThat(event.getSession().<Collection<String>>getProperty(MY_SESSION_LIST), is(expectedArray));
+        assertThat(event.getMessage().<Collection<String>>getProperty(MY_SESSION_LIST, PropertyScope.SESSION), is(expectedArray));
+    }
+
```
Comment (marcosnc) [inline tests/integration/src/test/java/org/mule/test/routing/ForEachSessionVarTestCase.java]: Add a test to ensure that event.getSession.getProperty(X) and event.getMessage.getProperty(X, PropertyScope.SESSION) point to the same map.

Comment (afelisatti) [inline tests/integration/src/test/java/org/mule/test/routing/ForEachSessionVarTestCase.java]: Good idea, I'll remove the property from the session and then verify it is no longer present in the message.

**Pull request closed without merging**